File: app/src/main/java/com/diamond/appcliente/adapters/ReservaAdapter.java
package com.diamond.appcliente.adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import com.diamond.appcliente.R;
import com.diamond.appcliente.dto.reserva.ReservaResponse;

import java.util.List;

public class ReservaAdapter extends RecyclerView.Adapter<ReservaAdapter.ReservaViewHolder> {

    private List<ReservaResponse> reservas;
    private Context context;

    public ReservaAdapter(List<ReservaResponse> reservas, Context context) {
        this.reservas = reservas;
        this.context = context;
    }

    @Override
    public ReservaViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.item_reserva, parent, false);
        return new ReservaViewHolder(view);
    }

    @Override
    public void onBindViewHolder(ReservaViewHolder holder, int position) {
        ReservaResponse reserva = reservas.get(position);
        holder.reservaId.setText(String.valueOf(reserva.getReservaId()));
        holder.barberoNombre.setText(reserva.getBarberoNombre());
        holder.horarioRango.setText(reserva.getHorarioRango());
        holder.estado.setText(reserva.getEstado());
    }

    @Override
    public int getItemCount() {
        return reservas.size();
    }

    public class ReservaViewHolder extends RecyclerView.ViewHolder {

        TextView reservaId, barberoNombre, horarioRango, estado;

        public ReservaViewHolder(View itemView) {
            super(itemView);
            reservaId = itemView.findViewById(R.id.reservaId);
            barberoNombre = itemView.findViewById(R.id.barberoNombre);
            horarioRango = itemView.findViewById(R.id.horarioRango);
        }
    }
}

File: app/src/main/java/com/diamond/appcliente/adapters/ReservaAdapter.kt
package com.diamond.appcliente.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.diamond.appcliente.R
import com.diamond.appcliente.dto.reserva.ReservaResponse

class ReservaAdapter(
    private val reservas: List<ReservaResponse>,
    private val context: Context
) : RecyclerView.Adapter<ReservaAdapter.ReservaViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ReservaViewHolder {
        val view = LayoutInflater.from(context).inflate(R.layout.item_reserva, parent, false)
        return ReservaViewHolder(view)
    }

    override fun onBindViewHolder(holder: ReservaViewHolder, position: Int) {
        val reserva = reservas[position]
        holder.reservaId.text = reserva.reservaId.toString()
        holder.barberoNombre.text = reserva.barberoNombre
        holder.horarioRango.text = reserva.horarioRango
        holder.estado.text = reserva.estado
    }

    override fun getItemCount() = reservas.size

    inner class ReservaViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val reservaId: TextView = itemView.findViewById(R.id.reservaId)
        val barberoNombre: TextView = itemView.findViewById(R.id.barberoNombre)
        val horarioRango: TextView = itemView.findViewById(R.id.horarioRango)
        val estado: TextView = itemView.findViewById(R.id.estado)
    }
}
